src/playback/internal/playbackcontroller.cpp

```
@@ -1386,6 +1391,104 @@ AudioOutputParams PlaybackController::trackOutputParams(const InstrumentTrackId&
+    }
+
+    auto buildAutomationEnvelope = [this](const AutomationCurve& curve) {
+        muse::audio::AutomationEnvelope envelope;
+        for (const auto& [tick, point] : curve) {
+            envelope.insert({ playedTickToSecs(tick), point.value });
+        }
+        return envelope;
+    };
+
+    if (rebuildVolume) {
+        const AutomationCurve& volumeCurve = automationData->curve(AutomationCurveKey::instrument(AutomationType::Volume,
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If Q_DECLARE_METATYPE is a macro then please configure it.

(unknownMacro)